src/SudoSu.php

```diff
@@ -23,15 +23,27 @@ public function __construct(Application $app, AuthManager $auth, SessionManager
         $this->session = $session;
     }
 
-    public function loginAsUser($userId, $currentUserId)
+    /**
+     * Log the sudosu user in and preserve the original user id if exists
+     *
+     * @param string $sudosuUserId
+     * @param string|null $originalUserId
+     * @return void
+     */
+    public function loginAsUser($sudosuUserId, $originalUserId)
     {
         $this->session->put('sudosu.has_sudoed', true);
-        $this->session->put($this->sessionKey, $currentUserId);
+        $this->session->put($this->sessionKey, $originalUserId);
 
-        $this->auth->loginUsingId($userId);
+        $this->auth->loginUsingId($sudosuUserId);
```

Comment:
Looks good, but can I ask what your reasoning is behind changing `$userId` to `$sudosuUserId` in `SudoSu::loginAsUser`?

Not disputing it, just curious!